doc/arch/adr-001-identity-and-authorization.md

```diff
@@ -0,0 +1,134 @@
+
+Most administrative tasks could still be done via the Django admin because the profile information would be there, but things like manual password resets would need to be done via the Auth0 console instead, which could introduce confusion for administrators.
+
+By integrating with an outside service we open ourselves up to needing to respond to changes in that services API, and there is also the risk that login could become impossible for a time if Auth0 goes down.
+
+Auth0 supports all major languages that we currently support in the app, and in general, has [a pretty extensive list of supported languages](https://auth0.com/docs/customize/internationalization-and-localization/universal-login-internationalization). However, if Boston has any immigrant enclaves that use less prominent languages that they want to add translations for, we might be out of luck.
+
+_Include the pros and cons of each option._
+
+### Django alone
+
+Pros:
+- Would follow an established pattern of Azavea applications stretching back many years.
+- Migration is relatively straightforward because accounts and profile data are all in the same place.
+- Essentially infinitely customizable (given sufficient budget)
+
+Cons:
+- Significant work to create full login / logout system from scratch, including integration with email service and frontend pages or components for users
+- Customization can be expensive in comparison to pushbutton solutions
+
+
+### Django + Auth0
+
+Pros:
+- In the ideal case, would cut down significantly on the work required to create the authentication system because we'd just need to implement the login/logout endpoints and Auth0 would handle the interface and emails.
+- Pushbutton support for other types of authentication like MFA and social logins (currently we haven't had requests to use either of those things, but it could happen in the future).
+
+Cons:
+- If we want to customize the domain or design of the login form then we'll need to pay, and we'll also need to integrate with an email provider.
+- Would split administrative tasks across two consoles depending on the task, which could be confusing for administrators.
+- Seems like it makes user imports somewhat more complex (in that we'd need two separate scripts and would have to make sure that everything linked up correctly)
+
+
+## Decision
+
+_What have we decided to do?_
+
+We have a (relatively) healthy budget and timeframe for completing this work, and part of the justification for undertaking the switch to Django was customizability and ease of administration. Given that, it probably makes the most sense to pursue a Django-only solution so that everything stays in one place and is as customizable as possible (even if that customization costs more in the long run).
+
+## Consequences
+
+- Customizations to the login pages will be relatively straightforward.
+- Customizations to the authentication process (e.g. adding social logins or MFA) will be relatively more complex.
```

Comment:
I'm sure there are other libraries available as well, but I've used `django-allauth` in the past for google / social media authentication and been pleasantly surprised at what the configuration entailed.

Reply:
I've used `django-social-auth` in the distant past and also had a good time, so yeah, we've probably got some decent libraries to choose from. I'll update the doc to make that clear.